### PR TITLE
【PaddlePaddle Hackathon 4】add paddle grid_sampler op

### PR DESCRIPTION
**Pull request by Patrick-Star125**
### Details:
add grid_sample operation in Paddle front end

### Reference:
- https://www.paddlepaddle.org.cn/documentation/docs/en/2.2/api/paddle/nn/functional/grid_sample_en.html#grid-sample
- https://github.com/openvinotoolkit/openvino/blob/a1510a5e5f5c664f6f005462c7fc5566f30bcad2/src/frontends/onnx/frontend/src/op/grid_sample.cpp

### unit-test

image

### CODE REVIEW

**Comment by Patrick-Star125** [comment]
@ceciliapeng2011 can you review this pr please?

**Comment by OpenVINO-dev-contest** [comment]
> @OpenVINO-dev-contest could you review this pr please?

Hi, Im ok with the implementation, Did you refer to openvino's onnx frontend ? if yes, pls add the link into this conversion and it will be helpful for other reviewer to understand your solution more easily. thanks

**Review by OpenVINO-dev-contest — APPROVED** [review]

**Review by xczhai — CHANGES_REQUESTED** [review]

**Comment by Patrick-Star125** [comment]
@xczhai 
1. the default value of `align_corners` should be true according to paddle [source code](https://github.com/PaddlePaddle/Paddle/blob/release/2.4/paddle/fluid/operators/grid_sampler_op.cc#L72)
2. test sample without optional inputs has been added
3. test samples with distinct shapes had been added

**Comment by xczhai** [comment]
> @xczhai
> 
> 1. the default value of `align_corners` should be true according to paddle [source code](https://github.com/PaddlePaddle/Paddle/blob/release/2.4/paddle/fluid/operators/grid_sampler_op.cc#L72)
> 2. test sample without optional inputs has been added
> 3. test samples with distinct shapes had been added

Hi @Patrick-Star125 , 
for dynamic shape test case, you can refer to https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/paddle/tests/test_models/gen_scripts/generate_dynamic_pool2d.py.
Why we appreciate `dynamic shape` is that `dynamic shape` plays an important role in the real deployment. Also, it can help understand the OpenVINO deeply and cover more scenario. Thanks.

**Comment by Patrick-Star125** [comment]
Got it, test sample with dynami shape has been added.

**Review by xczhai — APPROVED** [review]

File: src/frontends/paddle/src/op/grid_sampler.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+namespace ov {
+namespace frontend {
+namespace paddle {
+namespace op {
+NamedOutputs grid_sampler(const NodeContext& node) {
+    auto data = node.get_input("X");
+    auto grid = node.get_input("Grid");
+    default_opset::GridSample::Attributes attributes{};
+
```
Comment (xczhai) [inline src/frontends/paddle/src/op/grid_sampler.cpp:16]: please correct the code format here as log reports.

Comment (Patrick-Star125) [reply src/frontends/paddle/src/op/grid_sampler.cpp]: Done

**Review by ceciliapeng2011 — APPROVED** [review]

**Review by ceciliapeng2011 — APPROVED** [review]

**Pull request merged**